js/main.js

```diff
@@ -182,17 +182,40 @@ $form.addEventListener('submit', submitReview);
 function submitReview(event) {
   event.preventDefault();
   var review = {};
-  review = {
-    title: document.querySelector('#film-title').textContent,
-    img: document.querySelector('#film-banner').src,
-    textReview: $form.elements[0].value,
-    reviewId: data.nextReviewId
-  };
-
-  data.nextReviewId++;
-  data.reviews.unshift(review);
-  $ulElement2.prepend(createReviewList(review));
-  $form.reset();
+  if (data.editing === null) {
+    review = {
+      title: document.querySelector('#film-title').textContent,
+      img: document.querySelector('#film-banner').src,
+      textReview: $form.elements[0].value,
+      reviewId: data.nextReviewId
+    };
+
+    data.nextReviewId++;
+    data.reviews.unshift(review);
+    $ulElement2.prepend(createReviewList(review));
+    $form.reset();
+
+  } else if (data.editing !== null) {
+    review.title = document.querySelector('#film-title').textContent;
+    review.img = document.querySelector('#film-banner').src;
+    review.textReview = $form.elements[0].value;
+    review.reviewId = data.editing.reviewId;
+
+    for (var i = 0; i < data.reviews.length; i++) {
+      if (data.reviews[i].reviewId === review.reviewId) {
+        data.reviews[i] = review;
+      }
+    }
+    var reviewItem = document.querySelectorAll('#review-item');
+    for (i = 0; i < reviewItem.length; i++) {
+      var reviewItemId = JSON.parse(reviewItem[i].getAttribute('data-review-id'));
```

Comment:
You should use `parseInt`, not `JSON.parse`

Reply:
<img width="702" alt="Screen Shot 2022-08-04 at 2 59 25 PM" src="https://user-images.githubusercontent.com/97363006/182960316-0e9b56a8-402c-45a7-b854-cbc30fc58269.png">